scripts/pre_commit_linter.py

```diff
@@ -665,119 +682,170 @@ def _get_data(self, filepath, mode):
+            html_messages = html_lint_checks_manager.perform_all_lint_checks()
+            all_messages += html_messages
 
-    linting_processes.append(multiprocessing.Process(
-        target=_lint_py_files_for_python3_compatibility,
-        args=(
-            py_filepaths, py_result_for_python3_compatibility,
-            verbose_mode_enabled)))
+        if css_linter:
+            css_result = multiprocessing.Queue()
+            css_stdout = multiprocessing.Queue()
 
-    for process in linting_processes:
-        process.daemon = False
-        process.start()
+            _lint_css_files(
+                node_path,
+                stylelint_path,
+                config_path_for_oppia_css,
+                self.css_filepaths, css_stdout,
+                css_result, self.verbose_mode_enabled)
+
+            result_queues.append(css_result)
+            stdout_queues.append(css_stdout)
+
+        if html_linter or css_linter:
+            css_lint_checks_manager = CSSLintChecksManager(   # pylint: disable=no-value-for-parameter
+                self.verbose_mode_enabled)
+            css_messages = css_lint_checks_manager.perform_all_lint_checks()
+            all_messages += css_messages
+
+        if py_linter:
+            py_result = multiprocessing.Queue()
+            _lint_py_files(
+                config_pylint, config_pycodestyle, self.py_filepaths,
+                py_result, self.verbose_mode_enabled)
+
+            result_queues.append(py_result)
 
-    result_queues = [
-        js_and_ts_result, css_in_html_result, css_result, py_result,
-        py_result_for_python3_compatibility
-    ]
+            py_result_for_python3_compatibility = multiprocessing.Queue()
 
-    stdout_queus = [
-        css_in_html_stdout, css_stdout
-    ]
-    return linting_processes, result_queues, stdout_queus
+            _lint_py_files_for_python3_compatibility(
+                self.py_filepaths, py_result_for_python3_compatibility,
+                self.verbose_mode_enabled)
+
+            result_queues.append(py_result_for_python3_compatibility)
+
+            py_lint_checks_manager = PythonLintChecksManager( # pylint: disable=no-value-for-parameter
+                self.verbose_mode_enabled)
+
+            py_messages = py_lint_checks_manager.perform_all_lint_checks()
+
+            all_messages += py_messages
+
+        if other_linter:
+            other_lint_checks_manager = OtherLintChecksManager(   # pylint: disable=no-value-for-parameter
+                self.verbose_mode_enabled)
+            other_messages = other_lint_checks_manager.perform_all_lint_checks()
+            all_messages += other_messages
+
+        if codeowner_linter:
+            code_owner_message = _check_codeowner_file(
+                self.verbose_mode_enabled)
+            all_messages += code_owner_message
+
+        all_messages += _join_linting_process(
+            linting_processes, result_queues, stdout_queues)
```

Comment:
This is going to kick off all of the linting processes at once, which is what you don't want to do.

Reply:
Removed.